koji/collection-mash-split.py

```diff
@@ -328,34 +332,134 @@ def mash_split(self, whole_path, tmp_path, split_path, extras_path, mash_config,
             for pkg in glob.glob(os.path.join(path, "*.rpm")):
                 pkgs.add(kobo.rpmlib.parse_nvra(os.path.basename(pkg))["name"])
         for pkg in pkgs - all_from_comps:
-            self.logger.warning("%s not in %s" % (pkg, " nor ".join(compses)))
+            self.logger.warning("%s not in %s" % (pkg, " nor ".join(compses.keys())))
 
         self.check_tag_listing(mash_config)
 
-def main():
-    try:
-        version = sys.argv[1]
-    except IndexError:
-        version = "nightly"
 
-    s = MashSplit("/var/log/foreman-mash-split.log")
-    options = ["server"]
+def run_mashes(collection, git_tag, mashes):
+    log_file = "/var/log/{}-mash-split.log".format(collection)
+
     arches = ["x86_64"]
-    compses = ["comps-foreman-rhel7.xml"]
     whole_path = "/mnt/koji/releases/whole"
     tmp_path = "/mnt/koji/releases/tmp"
     split_path = "/mnt/koji/releases/split"
-    extras = "extras-foreman-rhel7"
     extras_path = "/mnt/koji/releases/extras"
 
-    mash_config = "foreman-{}-rhel7-dist".format(version)
-    if version == "nightly":
-        git_tag = "rpm/develop"
+    s = MashSplit(log_file)
+    for mash_config, compses, extras in mashes:
+        s.run_mash(whole_path, mash_config)
+        for extra in extras:
+            s.handle_extras(whole_path, mash_config, arches, extra, extras_path, git_tag)
+        s.handle_comps(whole_path, tmp_path, split_path, mash_config, arches, compses, git_tag)
+
+
+def main():
+    try:
+        collection = sys.argv[1]
+    except IndexError:
+        raise SystemExit("Usage: {} collection [version]".format(sys.argv[0]))
+
+    try:
+        version = sys.argv[2]
+    except IndexError:
+        version = "nightly"
+
+    if collection in ("foreman", "foreman-plugins"):
+        if version == "nightly":
+            git_tag = "rpm/develop"
+        else:
+            git_tag = "rpm/{}".format(version)
+
+        CONFIG["comps_path"] = "/mnt/koji/mash/comps/foreman"
+        if collection == 'foreman':
+            CONFIG["extras_baseloc"] = "%(tag)s:extras"
+            CONFIG["extras_path"] = "/mnt/koji/mash/extras/foreman"
+            extras = ["extras-foreman-rhel7"]
+        else:
+            extras = []
+
+        mashes = [
+            (
+                "{}-{}-rhel7-dist".format(collection, version),
```

Comment:
I wonder if this section would be more readable if we had smth like this:

```python
def mash_config_name(collection, version, suffix):
    return "{}-{}-{}".format(collection, version, suffix)

def comps_name(collection, suffix):
    return "comps-{}-{}.xml".format(collection, suffix)

(etc)
```

and then use it like:

```python
mashes = [
    (
        mash_config_name(collection, version, "rhel7-dist"),
        {comps_name(collection, "rhel7"), comps_path(collection, version, 'RHEL/7')},
        extras,
    ),
]
```